idunn/geocoder/nlu_client.py

```diff
@@ -68,19 +73,25 @@ def fuzzy_match(cls, query, response):
             return True
         return False
 
-    async def build_intention_category_place(self, tags_list, lang):
-        cat_query = next(t["phrase"] for t in tags_list if t.get("tag") == "cat")
-        city_query = next(t["phrase"] for t in tags_list if t.get("tag") == "city")
+    async def build_intention_category_place(
+        self, cat_query, place_query, lang, skip_classifier=False
+    ):
+        async def get_category():
+            if skip_classifier:
+                return None
+            else:
```

Comment:
No need for this else.

Reply:
Euh actually, you never leave this part of the code... if and else parts both return.